### PR TITLE
RM-001: Add optional default reconciliation hook for persisted values

### LINKED ISSUE #1: RM-001: Add optional default reconciliation hook for persisted values
## Summary
Persisted values can silently override newly shipped application defaults after deploy, even when the app intentionally changes global/default behavior.

## Current Behavior
- useMnemonicKey returns stored values when decode and validation succeeds.
- New defaults are only used when storage is absent or invalid.
- There is no first-class post-read reconciliation hook.

References:
- src/Mnemonic/use.ts (decodeForRead, reset, remove)

## Why This Matters
Existing users can remain on legacy defaults indefinitely, causing rollout inconsistency.

## Reproduction
1. Persist a key with an old default-like value.
2. Ship a new app version with a changed intended default.
3. Reload with existing storage.
4. Observe old persisted value still wins.

## Proposed Direction
Add optional reconcile hook in useMnemonicKey options, executed after decode and migration and before value exposure.

Example signature:
reconcile?: (value: T, context: { key: string; persistedVersion: number; latestVersion?: number }) => T

## Acceptance Criteria
- Selective field-level default enforcement without full key reset.
- Reconciled value persists once when changed.
- Works for schema-managed and codec-managed keys.
- Docs clarify when to use migration vs reconciliation.

### PR DESCRIPTION
**Pull request by thirtytwobits**
## Summary
- add a read-time `reconcile` hook to `useMnemonicKey`
- persist reconciled values when they differ from the stored payload
- add behavior-first coverage for codec and schema flows, including targeted mocked edge-case tests
- update README, guides, and generated API docs for reconciliation and current error handling

## Validation
- npm run docs
- npm run docs:site
- npm run lint
- npm run test
- npm run build

Closes #1

### CODE REVIEW

File: website/src/components/demo/ShoppingCart.tsx [hunk 1]
```diff
@@ -168,7 +168,7 @@ export function ShoppingCart() {
     }
 
     return (
-        <MnemonicProvider namespace="cart" storage={storage}>
+        <MnemonicProvider namespace="cart" storage={storage} enableDevTools>
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply website/src/components/demo/ShoppingCart.tsx]: Gated DevTools on the docs site behind a shared `enableDemoDevTools` flag so production-hosted demos do not expose the writable registry. Applied the shared gate to the page-level provider and the shopping cart provider, and updated the DevTools guide to state that hosted demos keep DevTools off in production.

File: src/Mnemonic/use.test.tsx [hunk 2]
```diff
@@ -178,6 +304,24 @@ describe("useMnemonicKey – codecs", () => {
         expect(errorSpy).toHaveBeenCalled();
         errorSpy.mockRestore();
     });
+
+    it("passes persisted version context into reconcile for codec-managed values", () => {
+        storage.store.set("ns.counter", env(JSON.stringify({ value: 5 }), 0));
+        const reconcile = vi.fn((value: { value: number }, context: { persistedVersion: number; latestVersion?: number }) => {
+            expect(context).toEqual({ persistedVersion: 0, key: "counter" });
+            return value;
+        });
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply src/Mnemonic/use.test.tsx]: Updated the test to type the callback context as the public `ReconcileContext` shape rather than an incomplete inline object, so the assertion now matches the API surface directly.

File: devtools-extension/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Quick start
+
+1. Open Chrome and go to `chrome://extensions`.
+2. Enable **Developer mode**.
+3. Click **Load unpacked**.
+4. Select this folder:
+
+   `/Users/thirtytwobits/workspace/github/thirtytwobits/react-mnemonic/devtools-extension`
+
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply devtools-extension/README.md]: Reworded the extension quick start to use the repo-relative `devtools-extension/` directory instead of a machine-specific absolute path.

File: src/Mnemonic/use.ts [hunk 4]
```diff
@@ -166,6 +166,144 @@ export function useMnemonicKey<T>(key: string, options: UseMnemonicKeyOptions<T>
...
+                if (nextRewriteRaw !== undefined) result.rewriteRaw = nextRewriteRaw;
+                if (nextPendingSchema !== undefined) result.pendingSchema = nextPendingSchema;
+                return result;
+            } catch (err) {
+                const typedErr =
+                    err instanceof SchemaError || err instanceof CodecError
+                        ? err
+                        : new SchemaError("RECONCILE_FAILED", `Reconciliation failed for key "${key}"`, err);
+                return { value: getFallback(typedErr) };
+            }
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply src/Mnemonic/use.ts]: Aligned the implementation with the documented contract: `reconcile` now preserves thrown `SchemaError` instances, and wraps all other failures (including `CodecError`) as `SchemaError("RECONCILE_FAILED")`. Added coverage for both the wrapped `CodecError` path and the preserved `SchemaError` path, and updated the docs/JSDoc wording to match.

File: src/Mnemonic/provider.tsx [hunk 5]
```diff
@@ -230,6 +284,130 @@ export function MnemonicProvider({
...
+            const finalizationRegistrySupported = hasFinalizationRegistry();
+            const globalWindow = window as any;
+            const rawExisting = globalWindow.__REACT_MNEMONIC_DEVTOOLS__;
+            const root: Record<string, any> =
+                rawExisting && typeof rawExisting === "object" ? rawExisting : {};
+
+            const reserved = new Set(["providers", "resolve", "list", "capabilities", "__meta"]);
+            for (const key of Object.keys(root)) {
+                if (!reserved.has(key)) {
+                    delete root[key];
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply src/Mnemonic/provider.tsx]: Hardened the legacy registry cleanup path by checking property descriptors before deletion and swallowing delete failures for hostile/non-configurable properties. Added a provider test that seeds an undeletable legacy field and verifies DevTools initialization still succeeds.

File: website/src/components/demo/SchemaPlayground.tsx [hunk 6]
```diff
@@ -849,6 +849,7 @@ export function SchemaPlayground() {
                 <MnemonicProvider
                     key={mountKey}
                     namespace={NAMESPACE}
+                    enableDevTools
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply website/src/components/demo/SchemaPlayground.tsx]: Applied the same shared production-safe demo gate here as well. `SchemaPlayground` now uses the shared `enableDemoDevTools` flag, so the docs site keeps DevTools disabled in production while local/dev builds still expose the registry for debugging.
